apps/ensadmin/src/components/identity/index.tsx

```diff
@@ -33,44 +42,61 @@ export function Identity({
   showAvatar = false,
   className,
 }: IdentityProps) {
-  const ensRootChainId = getENSRootChainId(namespaceId);
-
-  // Establish chainId, preferring user-supplied and defaulting to the ENS Root Chain Id.
-  const definedChainId = chainId ?? ensRootChainId;
+  if (chainId === undefined) {
+    // default to resolving the identity for `address` on the ENS Root Chain
+    // for the provided `namespaceId`.
+    chainId = getENSRootChainId(namespaceId);
+  }
 
-  // Lookup the primary name for address using ENSNode
+  // resolve the primary name for `address` on `chainId` using ENSNode
```

Comment:
if using the various `Identity` types i proposed, there could be a `useResolvedIdentity` hook like so:

```ts
const { identity, status } = useResolvedIdentity({ namespaceId, chainId, address });


function useResolvedIdentity({namespaceId, chainId, address}: UnresolvedIdentity) {
  const {data: name, ...query} = usePrimaryName({ address: identity.address, chainId: getResolvePrimaryNameChainIdParam(chainId, namespaceId) })

  return {
   ...query,
   identity: name !== undefined ? { namespaceId, chainId, address, name } : undefined
  }
}
```

and maybe that's clean?